htdocs/core/class/commonobject.class.php

```diff
@@ -6077,7 +6077,7 @@ public function fetch_optionals($rowid = null, $optionsArray = null)
 			$sql = "SELECT rowid";
 			foreach ($optionsArray as $name => $label) {
 				if (empty($extrafields->attributes[$this->table_element]['type'][$name]) || $extrafields->attributes[$this->table_element]['type'][$name] != 'separate') {
-					$sql .= ", ".$name;
+					$sql .= ", `".$this->db->escape($name)."`";
```

Comment:
Backtrick is not allowed in sql standard (it is a mysql extension)
If you escape data for string with db->escape, what you must use is simple quote ' (so the char thatvis escaped by the db->escape).

Reply:
The problem is that when a society uses an extrafield since the beginning, like a few years or more, it can be hard to change the field value without breaking anything.
From my point of view, it's quite hard to simply forbid reserved words, as their list evolves at each new version.

I'll keep testing and try to find the best way to fix it.

Reply:
In fact, the only way to pass through a reserved word is the use of backticks here, as `", '". $this->db->escape($name)."'"` won't make the reserved word as a field, but still as a reserved word.

The only way I found to make it work is the use of backticks. I don't know how to make it working correctly.

What do you think about the add of backticks only if we're in a MySQL configuration ? I could test `$this->db`

Reply:
Maybe you can add the table name, look

https://dev.mysql.com/doc/refman/8.0/en/keywords.html